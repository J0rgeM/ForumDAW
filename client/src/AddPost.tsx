import React, { useState } from "react";
// Library imports.
import axios, { AxiosResponse } from "axios";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";

function AddPost() {
    const [author, setAuthor] = useState("");
    const [body, setBody] = useState("");

    const addPost = async (e) => {
        e.preventDefault();
        try {
            const response: AxiosResponse = await axios.post(
                "http://localhost:8080/addPost",
                {
                    author: JSON.stringify(author),
                    body: JSON.stringify(body),
                    headers: {
                        "Content-type": "text/plain",
                    },
                }
            );
            window.location.href = "http://localhost:3000"
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <div>
            <Navbar />
            <div className="container text-center">
                <form onSubmit={(e) => addPost(e)} className="shadow p-3 mb-5 bg-white rounded">
                    <h1 className="mt-2">Post</h1>
                    <div className="mb-3">
                        <label form="inputTitle" className="form-label">
                            Message
                        </label>
                        <input
                            type="textarea"
                            className="form-control m-1"
                            id="inputPost"
                            onChange={(e) => setAuthor(e.target.value)}
                            placeholder="State your name"
                        ></input>
                        <input
                            type="textarea"
                            className="form-control m-1"
                            id="inputPost"
                            onChange={(e) => setBody(e.target.value)}
                            placeholder="Content of your post"
                        ></input>
                    </div>
                    <button
                        type="submit"
                        className="btn btn-dark">
                        Submit
                    </button>
                </form>
            </div>
            <Footer />
        </div>
    );
}

export default AddPost;
