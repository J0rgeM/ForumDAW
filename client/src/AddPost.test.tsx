import {describe, expect, it} from '@jest/globals';
import {renderWithRouter} from "./lib";
import AddPost from "./AddPost";

describe('AddPost', () => {
    it('match snapshot', () => {
        const { container} = renderWithRouter(<AddPost/>)
        expect(container).toMatchSnapshot()
    })
})