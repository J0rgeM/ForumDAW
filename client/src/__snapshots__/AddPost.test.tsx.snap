// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPost match snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
           BlogGIF
        </a>
        <button
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavDropdown"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse justify-content-end"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/addPost"
              >
                Add Post
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/"
              >
                List Posts
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container text-center"
    >
      <form
        class="shadow p-3 mb-5 bg-white rounded"
      >
        <h1
          class="mt-2"
        >
          Post
        </h1>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            form="inputTitle"
          >
            Message
          </label>
          <input
            class="form-control m-1"
            id="inputPost"
            placeholder="State your name"
            type="textarea"
          />
          <input
            class="form-control m-1"
            id="inputPost"
            placeholder="Content of your post"
            type="textarea"
          />
        </div>
        <button
          class="btn btn-dark"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="footer text-center fixed-bottom text-white bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <p
            class="text-center"
          >
            Â© 2024242 DAW
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
